File: src/colorcss.go
package pdfjet

/**
 * colorcss.go
 *
Copyright 2023 Innovatics Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

func NewColorCSS() map[string]uint32 {
	color := make(map[string]uint32)
	color["aliceblue"] = 0xf0f8ff
	color["transparent"] = 0x80000000
	color["aliceblue"] = 0xf0f8ff
	color["antiquewhite"] = 0xfaebd7
	color["aqua"] = 0x00ffff
	color["aquamarine"] = 0x7fffd4
	color["azure"] = 0xf0ffff
	color["beige"] = 0xf5f5dc
	color["bisque"] = 0xffe4c4
	color["black"] = 0x000000
	color["blanchedalmond"] = 0xffebcd
	color["blue"] = 0x0000ff
	color["blueviolet"] = 0x8a2be2
	color["brown"] = 0xa52a2a
	color["burlywood"] = 0xdeb887
	color["cadetblue"] = 0x5f9ea0
	color["chartreuse"] = 0x7fff00
	color["chocolate"] = 0xd2691e
	color["coral"] = 0xff7f50
	color["cornflowerblue"] = 0x6495ed
	color["cornsilk"] = 0xfff8dc
	color["crimson"] = 0xdc143c
	color["cyan"] = 0x00ffff
	color["darkblue"] = 0x00008b
	color["darkcyan"] = 0x008b8b
	color["darkgoldenrod"] = 0xb8860b
	color["darkgray"] = 0xa9a9a9
	color["darkgreen"] = 0x006400
	color["darkgrey"] = 0xa9a9a9
	color["darkkhaki"] = 0xbdb76b
	color["darkmagenta"] = 0x8b008b
	color["darkolivegreen"] = 0x556b2f
	color["darkorange"] = 0xff8c00
	color["darkorchid"] = 0x9932cc
	color["darkred"] = 0x8b0000
	color["darksalmon"] = 0xe9967a
	color["darkseagreen"] = 0x8fbc8f
	color["darkslateblue"] = 0x483d8b
	color["darkslategray"] = 0x2f4f4f
	color["darkslategrey"] = 0x2f4f4f
	color["darkturquoise"] = 0x00ced1
	color["darkviolet"] = 0x9400d3
	color["deeppink"] = 0xff1493
	color["deepskyblue"] = 0x00bfff
	color["dimgray"] = 0x696969
	color["dimgrey"] = 0x696969
	color["dodgerblue"] = 0x1e90ff
	color["firebrick"] = 0xb22222
	color["floralwhite"] = 0xfffaf0
	color["forestgreen"] = 0x228b22
	color["fuchsia"] = 0xff00ff
	color["gainsboro"] = 0xdcdcdc
	color["ghostwhite"] = 0xf8f8ff
	color["gold"] = 0xffd700
	color["goldenrod"] = 0xdaa520
	color["gray"] = 0x808080
	color["green"] = 0x008000
	color["greenyellow"] = 0xadff2f
	color["grey"] = 0x808080
	color["honeydew"] = 0xf0fff0
	color["hotpink"] = 0xff69b4
	color["indianred"] = 0xcd5c5c
	color["indigo"] = 0x4b0082
	color["ivory"] = 0xfffff0
	color["khaki"] = 0xf0e68c
	color["lavender"] = 0xe6e6fa
	color["lavenderblush"] = 0xfff0f5
	color["lawngreen"] = 0x7cfc00
	color["lemonchiffon"] = 0xfffacd
	color["lightblue"] = 0xadd8e6
	color["lightcoral"] = 0xf08080
	color["lightcyan"] = 0xe0ffff
	color["lightgoldenrodyellow"] = 0xfafad2
	color["lightgray"] = 0xd3d3d3
	color["lightgreen"] = 0x90ee90
	color["lightgrey"] = 0xd3d3d3
	color["lightpink"] = 0xffb6c1
	color["lightsalmon"] = 0xffa07a
	color["lightseagreen"] = 0x20b2aa
	color["lightskyblue"] = 0x87cefa
	color["lightslategray"] = 0x778899
	color["lightslategrey"] = 0x778899
	color["lightsteelblue"] = 0xb0c4de
	color["lightyellow"] = 0xffffe0
	color["lime"] = 0x00ff00
	color["limegreen"] = 0x32cd32
	color["linen"] = 0xfaf0e6
	color["magenta"] = 0xff00ff
	color["maroon"] = 0x800000
	color["mediumaquamarine"] = 0x66cdaa
	color["mediumblue"] = 0x0000cd
	color["mediumorchid"] = 0xba55d3
	color["mediumpurple"] = 0x9370db
	color["mediumseagreen"] = 0x3cb371
	color["mediumslateblue"] = 0x7b68ee
	color["mediumspringgreen"] = 0x00fa9a
	color["mediumturquoise"] = 0x48d1cc
	color["mediumvioletred"] = 0xc71585
	color["midnightblue"] = 0x191970
	color["mintcream"] = 0xf5fffa
	color["mistyrose"] = 0xffe4e1
	color["moccasin"] = 0xffe4b5
	color["navajowhite"] = 0xffdead
	color["navy"] = 0x000080
	color["oldlace"] = 0xfdf5e6
	color["olive"] = 0x808000
	color["olivedrab"] = 0x6b8e23
	color["orange"] = 0xffa500
	color["orangered"] = 0xff4500
	color["orchid"] = 0xda70d6
	color["palegoldenrod"] = 0xeee8aa
	color["palegreen"] = 0x98fb98
	color["paleturquoise"] = 0xafeeee
	color["palevioletred"] = 0xdb7093
	color["papayawhip"] = 0xffefd5
	color["peachpuff"] = 0xffdab9
	color["peru"] = 0xcd853f
	color["pink"] = 0xffc0cb
	color["plum"] = 0xdda0dd
	color["powderblue"] = 0xb0e0e6
	color["purple"] = 0x800080
	color["red"] = 0xff0000
	color["rosybrown"] = 0xbc8f8f
	color["royalblue"] = 0x4169e1
	color["saddlebrown"] = 0x8b4513
	color["salmon"] = 0xfa8072
	color["sandybrown"] = 0xf4a460
	color["seagreen"] = 0x2e8b57
	color["seashell"] = 0xfff5ee
	color["sienna"] = 0xa0522d
	color["silver"] = 0xc0c0c0
	color["skyblue"] = 0x87ceeb
	color["slateblue"] = 0x6a5acd
	color["slategray"] = 0x708090
	color["slategrey"] = 0x708090
	color["snow"] = 0xfffafa
	color["springgreen"] = 0x00ff7f
	color["steelblue"] = 0x4682b4
	color["tan"] = 0xd2b48c
	color["teal"] = 0x008080
	color["thistle"] = 0xd8bfd8
	color["tomato"] = 0xff6347
	color["turquoise"] = 0x40e0d0
	color["violet"] = 0xee82ee
	color["wheat"] = 0xf5deb3
	color["white"] = 0xffffff
	color["whitesmoke"] = 0xf5f5f5
	color["yellow"] = 0xffff00
	color["yellowgreen"] = 0x9acd32
	color["oldgloryred"] = 0xb22234
	color["oldgloryblue"] = 0x3c3b6e
	return color
}

File: Sources/PDFjet/ColorCSS.swift
/**
 *  ColorCSS.swift
 *
Copyright 2023 Innovatics Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/


/**
 * Used to specify the pen and brush colors.
 * @see <a href="http://www.w3.org/TR/css3-color/#svg-color">http://www.w3.org/TR/css3-color/#svg-color</a>
 *
 */
public class ColorCSS {
    public let transparent: UInt32 = 0x80000000

    public let aliceblue: UInt32 = 0xf0f8ff
    public let antiquewhite: UInt32 = 0xfaebd7
    public let aqua: UInt32 = 0x00ffff
    public let aquamarine: UInt32 = 0x7fffd4
    public let azure: UInt32 = 0xf0ffff
    public let beige: UInt32 = 0xf5f5dc
    public let bisque: UInt32 = 0xffe4c4
    public let black: UInt32 = 0x000000
    public let blanchedalmond: UInt32 = 0xffebcd
    public let blue: UInt32 = 0x0000ff
    public let blueviolet: UInt32 = 0x8a2be2
    public let brown: UInt32 = 0xa52a2a
    public let burlywood: UInt32 = 0xdeb887
    public let cadetblue: UInt32 = 0x5f9ea0
    public let chartreuse: UInt32 = 0x7fff00
    public let chocolate: UInt32 = 0xd2691e
    public let coral: UInt32 = 0xff7f50
    public let cornflowerblue: UInt32 = 0x6495ed
    public let cornsilk: UInt32 = 0xfff8dc
    public let crimson: UInt32 = 0xdc143c
    public let cyan: UInt32 = 0x00ffff
    public let darkblue: UInt32 = 0x00008b
    public let darkcyan: UInt32 = 0x008b8b
    public let darkgoldenrod: UInt32 = 0xb8860b
    public let darkgray: UInt32 = 0xa9a9a9
    public let darkgreen: UInt32 = 0x006400
    public let darkgrey: UInt32 = 0xa9a9a9
    public let darkkhaki: UInt32 = 0xbdb76b
    public let darkmagenta: UInt32 = 0x8b008b
    public let darkolivegreen: UInt32 = 0x556b2f
    public let darkorange: UInt32 = 0xff8c00
    public let darkorchid: UInt32 = 0x9932cc
    public let darkred: UInt32 = 0x8b0000
    public let darksalmon: UInt32 = 0xe9967a
    public let darkseagreen: UInt32 = 0x8fbc8f
    public let darkslateblue: UInt32 = 0x483d8b
    public let darkslategray: UInt32 = 0x2f4f4f
    public let darkslategrey: UInt32 = 0x2f4f4f
    public let darkturquoise: UInt32 = 0x00ced1
    public let darkviolet: UInt32 = 0x9400d3
    public let deeppink: UInt32 = 0xff1493
    public let deepskyblue: UInt32 = 0x00bfff
    public let dimgray: UInt32 = 0x696969
    public let dimgrey: UInt32 = 0x696969
    public let dodgerblue: UInt32 = 0x1e90ff
    public let firebrick: UInt32 = 0xb22222
    public let floralwhite: UInt32 = 0xfffaf0
    public let forestgreen: UInt32 = 0x228b22
    public let fuchsia: UInt32 = 0xff00ff
    public let gainsboro: UInt32 = 0xdcdcdc
    public let ghostwhite: UInt32 = 0xf8f8ff
    public let gold: UInt32 = 0xffd700
    public let goldenrod: UInt32 = 0xdaa520
    public let gray: UInt32 = 0x808080
    public let green: UInt32 = 0x008000
    public let greenyellow: UInt32 = 0xadff2f
    public let grey: UInt32 = 0x808080
    public let honeydew: UInt32 = 0xf0fff0
    public let hotpink: UInt32 = 0xff69b4
    public let indianred: UInt32 = 0xcd5c5c
    public let indigo: UInt32 = 0x4b0082
    public let ivory: UInt32 = 0xfffff0
    public let khaki: UInt32 = 0xf0e68c
    public let lavender: UInt32 = 0xe6e6fa
    public let lavenderblush: UInt32 = 0xfff0f5
    public let lawngreen: UInt32 = 0x7cfc00
    public let lemonchiffon: UInt32 = 0xfffacd
    public let lightblue: UInt32 = 0xadd8e6
    public let lightcoral: UInt32 = 0xf08080
    public let lightcyan: UInt32 = 0xe0ffff
    public let lightgoldenrodyellow: UInt32 = 0xfafad2
    public let lightgray: UInt32 = 0xd3d3d3
    public let lightgreen: UInt32 = 0x90ee90
    public let lightgrey: UInt32 = 0xd3d3d3
    public let lightpink: UInt32 = 0xffb6c1
    public let lightsalmon: UInt32 = 0xffa07a
    public let lightseagreen: UInt32 = 0x20b2aa
    public let lightskyblue: UInt32 = 0x87cefa
    public let lightslategray: UInt32 = 0x778899
    public let lightslategrey: UInt32 = 0x778899
    public let lightsteelblue: UInt32 = 0xb0c4de
    public let lightyellow: UInt32 = 0xffffe0
    public let lime: UInt32 = 0x00ff00
    public let limegreen: UInt32 = 0x32cd32
    public let linen: UInt32 = 0xfaf0e6
    public let magenta: UInt32 = 0xff00ff
    public let maroon: UInt32 = 0x800000
    public let mediumaquamarine: UInt32 = 0x66cdaa
    public let mediumblue: UInt32 = 0x0000cd
    public let mediumorchid: UInt32 = 0xba55d3
    public let mediumpurple: UInt32 = 0x9370db
    public let mediumseagreen: UInt32 = 0x3cb371
    public let mediumslateblue: UInt32 = 0x7b68ee
    public let mediumspringgreen: UInt32 = 0x00fa9a
    public let mediumturquoise: UInt32 = 0x48d1cc
    public let mediumvioletred: UInt32 = 0xc71585
    public let midnightblue: UInt32 = 0x191970
    public let mintcream: UInt32 = 0xf5fffa
    public let mistyrose: UInt32 = 0xffe4e1
    public let moccasin: UInt32 = 0xffe4b5
    public let navajowhite: UInt32 = 0xffdead
    public let navy: UInt32 = 0x000080
    public let oldlace: UInt32 = 0xfdf5e6
    public let olive: UInt32 = 0x808000
    public let olivedrab: UInt32 = 0x6b8e23
    public let orange: UInt32 = 0xffa500
    public let orangered: UInt32 = 0xff4500
    public let orchid: UInt32 = 0xda70d6
    public let palegoldenrod: UInt32 = 0xeee8aa
    public let palegreen: UInt32 = 0x98fb98
    public let paleturquoise: UInt32 = 0xafeeee
    public let palevioletred: UInt32 = 0xdb7093
    public let papayawhip: UInt32 = 0xffefd5
    public let peachpuff: UInt32 = 0xffdab9
    public let peru: UInt32 = 0xcd853f
    public let pink: UInt32 = 0xffc0cb
    public let plum: UInt32 = 0xdda0dd
    public let powderblue: UInt32 = 0xb0e0e6
    public let purple: UInt32 = 0x800080
    public let red: UInt32 = 0xff0000
    public let rosybrown: UInt32 = 0xbc8f8f
    public let royalblue: UInt32 = 0x4169e1
    public let saddlebrown: UInt32 = 0x8b4513
    public let salmon: UInt32 = 0xfa8072
    public let sandybrown: UInt32 = 0xf4a460
    public let seagreen: UInt32 = 0x2e8b57
    public let seashell: UInt32 = 0xfff5ee
    public let sienna: UInt32 = 0xa0522d
    public let silver: UInt32 = 0xc0c0c0
    public let skyblue: UInt32 = 0x87ceeb
    public let slateblue: UInt32 = 0x6a5acd
    public let slategray: UInt32 = 0x708090
    public let slategrey: UInt32 = 0x708090
    public let snow: UInt32 = 0xfffafa
    public let springgreen: UInt32 = 0x00ff7f
    public let steelblue: UInt32 = 0x4682b4
    public let tan: UInt32 = 0xd2b48c
    public let teal: UInt32 = 0x008080
    public let thistle: UInt32 = 0xd8bfd8
    public let tomato: UInt32 = 0xff6347
    public let turquoise: UInt32 = 0x40e0d0
    public let violet: UInt32 = 0xee82ee
    public let wheat: UInt32 = 0xf5deb3
    public let white: UInt32 = 0xffffff
    public let whitesmoke: UInt32 = 0xf5f5f5
    public let yellow: UInt32 = 0xffff00
    public let yellowgreen: UInt32 = 0x9acd32

    public let oldgloryred: UInt32 = 0xb22234
    public let oldgloryblue: UInt32 = 0x3c3b6e
}
